podoglyph/gradebook
# Development

#### What does this PR do?
This PR sets up the initial framework for sessions including routes, controller, views, and feature spec for the Teacher role to login. To that end, it creates the Teacher Dashboard route, which will display a given teacher's courses and students.

#### Where should the reviewer start?
spec/features/users/teacher_can_login_spec

#### How should this be manually tested?
Run the server and login with default User with Teacher role: first_name: Don, last_name: Quixote, password: password. This should route to a barebones Teacher Dashboard.

#### Any background context you want to provide?
The purpose of these additions is primarily to setup a pattern upon which we can create Admin and Student dashboards.

#### What are the relevant tickets?

#### Screenshots (if appropriate)

#### Questions:

  - Do Migrations Need to be ran?
* Yes

  - Do Environment Variables need to be set?
* No
  - Any other deploy steps?
* No
